### ISSUE TITLE
what virtual tables are in rqlite/ sqlite ? 

### ISSUE THREAD

**Issue by Dorsosiv** [rqlite/rqlite#843]
Hi @otoolep , We tried to see what is making our the rqlite consume more memory , and we tried to find a table called "dbstat" which holds table size statistics for a sqlite database, but it was not there. could you specify if there are any virtual tables and what are they ? 

we tried also getting the data from the .expvar command but it does not have any info about table size.

Thanks :)

**Comment by otoolep** [comment]
That's because rqlite didn't compile SQLite with the required option enabled, `SQLITE_ENABLE_DBSTAT_VTAB`, enabled.

Run to confirm:

`curl -s -G 'localhost:4001/db/query?timings&pretty' --data-urlencode 'q=PRAGMA compile_options;'`

I can certainly look into enabling it if you like. It sounds useful.

**Comment by otoolep** [comment]
If you like, you can rebuild rqlite with `dbstat` support enabled. Build rqlite as follows:

```bash
cd rqlite/rqlite/cmd/rqlited
CGO_CFLAGS="-DSQLITE_ENABLE_DBSTAT_VTAB" go build .
```

Then, you can confirm `dbstat` is available like so:

```bash
$ curl -s -G 'localhost:4001/db/query?timings&pretty' --data-urlencode 'q=PRAGMA compile_options;'
{
    "results": [
        {
            "columns": [
                "compile_options"
            ],
            "types": [
                ""
            ],
            "values": [
                [
                    "COMPILER=gcc-7.5.0"
                ],
                [
                    "DEFAULT_WAL_SYNCHRONOUS=1"
                ],
                [
                    "ENABLE_DBSTAT_VTAB"
                ],
                [
                    "ENABLE_FTS3"
                ],
                [
                    "ENABLE_FTS3_PARENTHESIS"
                ],
                [
                    "ENABLE_RTREE"
                ],
                [
                    "ENABLE_UPDATE_DELETE_LIMIT"
                ],
                [
                    "OMIT_DEPRECATED"
                ],
                [
                    "SYSTEM_MALLOC"
                ],
                [
                    "THREADSAFE=1"
                ]
            ],
            "time": 0.000099703
        }
    ],
    "time": 0.000539985
}
```

Let me know if this works for you, and I'll consider enabling it by default in the next release.

**Comment by otoolep** [comment]
Fixed in 6.1.0.

**Comment by Dorsosiv** [comment]
great Ill check that :)
